functorch/functorch/experimental/cond.py

```diff
@@ -0,0 +1,105 @@
+import torch
+from torch._C import DispatchKey, DispatchKeySet, ExcludeDispatchKeyGuard
+from functorch.experimental.ops import PyOperator
+from torch.utils._pytree import tree_flatten
+from torch.fx.experimental.proxy_tensor import ProxyTensor, get_isolated_graphmodule
+import torch.utils._pytree as pytree
+
+"""
+We're going to define a `cond` operation.
+In order to do this, we need implementations for each of the dispatch keys.
+"""
+from contextlib import contextmanager
+
+# TODO(voz): Move out somewhere else once other py dispatched ops need it
+@contextmanager
+def suspend_mode(mode):
```

Comment:
Is `no_dispatch()` what you're looking for? https://github.com/pytorch/pytorch/blob/master/torch/utils/_mode_utils.py#L134

Reply:
Implementation wise, no, this ones sets the mode to None, then flips it back to what it was.

no_dispatch() installs 2 guards on the dispatcher (`Python` and `PythonTLSSnapshot`) and then deletes them.

Outcome wise? I dont know the C++ dispatcher well enough to say. I do know that I explicitly need this specific node suspended for the duration of the call, so even if they had the same outcome, I'd kind of want to keep this if only for clarity (as long as it doesn't have some unexpected side effects / do the wrong thing).

Reply:
Suspend mode is the right thing here, we have some logic in the C++ dispatcher (that I can't find anymore) that suspends just the mode right before dispatching to it